packages/pieces/src/lib/apps/twilio/trigger/new-incoming-sms.ts

```diff
@@ -0,0 +1,103 @@
+        "api_version": "2010-04-01",
+        "subresource_uris": {
+            "media": "/2010-04-01/Accounts/ACc0ea716d61fe90d78a69a3de71d45619/Messages/SM8c3920d3f2ac481ba83e639a69dadd63/Media.json",
+            "feedback": "/2010-04-01/Accounts/ACc0ea716d61fe90d78a69a3de71d45619/Messages/SM8c3920d3f2ac481ba83e639a69dadd63/Feedback.json"
+        }
+    },
+    type: TriggerStrategy.POLLING,
+    async onEnable(context) {
+        const { phone_number } = context.propsValue;
+        const account_sid = context.propsValue['authentication']!['username']!;
+        const auth_token = context.propsValue['authentication']!['password']!;
+        const response = await callTwilioApi<MessagePaginationResponse>(HttpMethod.GET, `Messages.json?PageSize=20&To=${phone_number}`, { account_sid, auth_token, }, {});
+        await context.store?.put<LastMessage>('_new_incoming_sms_trigger', {
+            lastMessageId: response.body.messages.length === 0 ? null : response.body.messages[0].sid,
+        });
+    },
+    async onDisable(context) {
+        await context.store?.put('_new_incoming_sms_trigger', null);
+    },
+    async run(context) {
+        const account_sid = context.propsValue['authentication']!['username']!;
+        const auth_token = context.propsValue['authentication']!['password']!;
+        const newMessages: unknown[] = [];
+        const lastMessage = await context.store.get<LastMessage>('_new_incoming_sms_trigger');
+        let currentUri: string | null = `2010-04-01/Accounts/${account_sid}/Messages.json?PageSize=20&To=${context.propsValue.phone_number}`;
+        let firstMessageId = undefined;
+        while (currentUri !== undefined && currentUri !== null) {
+            const res: HttpResponse<MessagePaginationResponse> = (await httpClient.sendRequest<MessagePaginationResponse>({
+                method: HttpMethod.GET,
+                url: `https://api.twilio.com/${currentUri}`,
+                authentication: {
+                    type: AuthenticationType.BASIC,
+                    username: account_sid,
+                    password: auth_token,
+                }
+            }));
+            const messages = res.body.messages;
+            if (!firstMessageId && messages.length > 0) {
+                firstMessageId = messages[0].sid;
+            }
+            currentUri = res.body.next_page_uri;
+            for (let i = 0; i < messages.length; i++) {
+                const message = messages[i];
+                if (message.sid === lastMessage?.lastMessageId) {
+                    currentUri = null;
+                    break;
+                }
+                if (message.direction === 'inbound') {
+                    newMessages.push(message);
+                }
+            }
+        }
+        await context.store?.put<LastMessage>('_new_incoming_sms_trigger', {
+            lastMessageId: firstMessageId ?? lastMessage!.lastMessageId,
+        });
+        return newMessages;
+    }
+});
```

Comment:
Twilio's API [docs](https://www.twilio.com/docs/sms/api/message-resource#read-multiple-message-resources), says there's a `DateSent` query param that can be used to filter messages.

Can this query param be used to simplify the implementation for this trigger?

Reply:
Sadly no, They are sorted in descending order, I will have to traverse backward until reach certain id.